File: src/main/java/me/iankit/simplechat/chat/ChatController.java
package me.iankit.simplechat.chat;

import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;
import reactor.core.scheduler.Schedulers;

import java.time.LocalDateTime;

@RequestMapping("chat")
@RestController
public class ChatController {
    private final ChatDao chatDao;

    public ChatController(ChatDao chatDao) {
        this.chatDao = chatDao;
    }

    @GetMapping(value = "/id/{chatId}", produces = MediaType.TEXT_EVENT_STREAM_VALUE)
    public Flux<Chat> getMessages(@PathVariable Integer chatId) {
        return chatDao.findByChatId(chatId)
                .subscribeOn(Schedulers.boundedElastic());
    }

    @PostMapping
    public Mono<Chat> newMessage(@RequestBody Chat chat) {
        chat.setCreatedAt(LocalDateTime.now());
        return chatDao.save(chat);
    }
}

File: src/main/java/me/iankit/simplechat/chat/ChatController.kt
package me.iankit.simplechat.chat

import lombok.extern.slf4j.Slf4j
import org.jetbrains.annotations.NotNull
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.*
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import reactor.core.scheduler.Schedulers

@RequestMapping("chats")
@RestController
@Slf4j
class ChatController(private val chatDao: ChatDao) {
    @GetMapping(value = ["/{groupId}"], produces = [MediaType.TEXT_EVENT_STREAM_VALUE])
    fun getMessages(@PathVariable @NotNull groupId: String): Flux<Chat> {
        return chatDao.findByGroupId(groupId)
            .log()
            .subscribeOn(Schedulers.boundedElastic())
    }

    @PostMapping
    fun newMessage(@RequestBody chat: Chat): Mono<Chat> {
        return chatDao.save(chat)
    }
}